compiler-core/src/build/project_compiler.rs

```diff
@@ -96,6 +96,10 @@ where
         self.options.mode
     }
 
+    pub fn warnings(&mut self) -> Vec<Warning> {
```

Comment:
Could you call this `take_warnings` please? To make it clear that it mutates the state of the compiler.